web/pages/[username]/calibration.tsx

```diff
@@ -205,8 +205,26 @@ const getCalibrationPoints = (betsData: [Contract, LimitBet[]][]) => {
     if (resolution !== 'YES' && resolution !== 'NO') continue
     const resolvedYES = resolution === 'YES'
 
-    for (const bet of bets as Bet[]) {
-      if (bet.amount < 0) continue // skip sales
+    let currentPosition = 0
+
+    // bets is reversed in place to track user current position
+    for (const bet of bets.slice().reverse() as Bet[]) {
+      const betSign = bet.outcome === 'YES' ? 1 : -1
+      const nextPosition = currentPosition + bet.shares * betSign
+
+      // skip explicit and exclusive implicit sales
+      if (bet.amount < 0 || ((Math.sign(currentPosition) !== betSign) && (Math.abs(currentPosition) >= Math.abs(bet.shares)))) {
+        currentPosition = nextPosition  // update position before continuing
+        continue
+      }
+
+      // set bet weight, adjusted for any partial implicit sale
+      let w = bet.shares
+      if (Math.sign(currentPosition) !== betSign && Math.abs(currentPosition) < Math.abs(bet.shares)) {
+        w = nextPosition
```

Comment:
This should be the abs value of next pos; the weight should always be positive.

Other than that, it looks good. I'll make this fix and merge it in...soon...